hw/ip/sram_ctrl/doc/dv/index.md

```diff
@@ -39,32 +41,70 @@ The following utilities provide generic helper tasks and functions to perform ac
 `sram_ctrl_env_pkg`. Some of them in use are:
 ```systemverilog
-[list a few parameters, types & methods; no need to mention all]
+// number of bits in a full KDI transactions
+parameter int KDI_DATA_SIZE = 1 + otp_ctrl_pkg::SramKeyWidth + otp_ctrl_pkg::SramNonceWidth;
+
+// a completed KDI transaction takes 4 cycles to be fully synchronized into the
+// SRAM clock domain
+parameter int KDI_PROPAGATION_CYCLES = 4;
+
+// a LC escalation request needs 3 cycles to be fully propagated through the DUT
+parameter int LC_ESCALATION_PROPAGATION_CYCLES = 3;
+
+typedef virtual mem_bkdr_if #(.MEM_PARITY(1)) mem_bkdr_vif;
+
+typedef enum bit {
+  SramCtrlRenewScrKey = 0,
+  SramCtrlInit        = 1
+} sram_ctrl_e;
+
+typedef enum bit [1:0] {
+  SramCtrlError           = 0,
+  SramCtrlEscalated       = 1,
+  SramCtrlScrKeyValid     = 2,
+  SramCtrlScrKeySeedValid = 3
+} sram_ctrl_status_e;
 ```
 ### TL_agent
 SRAM_CTRL testbench instantiates (already handled in CIP base env) [tl_agent]({{< relref "hw/dv/sv/tl_agent/README.md" >}})
-which provides the ability to drive and independently monitor random traffic via
-TL host interface into SRAM_CTRL device.
+which provides the ability to drive and independently monitor random traffic via TL host interface into SRAM_CTRL device.
+
+### Alert agents
+The SRAM_CTRL testbench instantiates 2 [alert agents]({{< relref "hw/dv/sv/alert_esc_agent/README.md" >}}) for:
+* fatal_intg_error - signals a transmission integrity error
+* fatal_parity_error - signals an internal parity error in the memory primitive
 
-### UVC/agent 1
-[Describe here or add link to its README]
+The alert agents provide the ability to independently drive and monitor both alert handshakes.
 
-### UVC/agent 2
-[Describe here or add link to its README]
+### SRAM_CTRL_LC_IF
+The SRAM_CTRL IP has a simple sideband interface to the LC_CTRL to receive escalation requests.
+This interface initializes the escalation bus and utilizes a simple task to drive escalation requests.
+
+### SRAM_CTRL_EXEC_IF
+The SRAM_CTRL IP has an interface to enable instruction execution from SRAM - allowing the Icache to fetch instruction data from the SRAM for the CPU.
+This interface contains the necessary Lifecycle and OTP structs to enable and disable this functionality.
 
 ### UVM RAL Model
 The SRAM_CTRL RAL model is created with the [`ralgen`]({{< relref "hw/dv/tools/ralgen/README.md" >}}) FuseSoC generator script automatically when the simulation is at the build stage.
 
 It can be created manually by invoking [`regtool`]({{< relref "util/reggen/README.md" >}}):
 
+<!-- TODO add description of second RAL model once implemented -->
+
 ### Reference models
-[Describe reference models in use if applicable, example: SHA256/HMAC]
+The SRAM_CTRL testbench uses a SystemVerilog golden model of its address and data [scrambling]({{< relref "hw/dv/sv/mem_bkdr_if" >}}) to check that encryption and decryption are performed correctly.
```

Comment:
Just a question, I do not know if it is correct. `mem_bkdr_if` sounds more like a memory backdoor interface rather a reference models. But I do understand the SRAM encrypt/decrypt methods are checked inside that interface...

Reply:
hmm yes that's a good point. I should probably explain briefly that the golden model is within the mem_bkdr_if.